rest_framework_dataclasses/union_serializer.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+import inspect
+from typing import Optional
+
+from rest_framework import serializers
+from rest_framework.fields import empty
+
+from rest_framework_dataclasses.import_utils import import_polymorphic_serializer
+
+PolymorphicSerializer = import_polymorphic_serializer()
+
+
+class InlinePolymorphicSerializer(PolymorphicSerializer):
+    model_serializer_mapping = None
+    resource_type_field_name = "resourcetype"
+
+    class Meta:
+        ref_name: Optional[str] = None
+
+    """
+    This class utlizes PolymorphicSerializer from rest_polymorphic packageto support
+    any class object (not limited to django models).
+
+    Creating PolymorphicSerializer requires declaring a new class derived from it with "model_serializer_mapping"
+    classvar.
+    We could create dynamically new class derived from PolymorphicSerializer with "model_serializer_mapping" classvar,
+    but in case that we have 2 classes with the same name (for example, same union in two different serializer)
+    drf-spectacular warns that there is a conflict (2 different classes with the same name).
+    In order to avoid this warning we create new class that let us get the "model_serializer_mapping"
+    classvar (and few others classvars) from the constructor and actually make them instance members.
```

Comment:
I'm not sure I fully understand this reasoning: I get that `drf-spectacular` doesn't like it if there's two serializer classes with the same name, but I don't see how creating this custom class prevents that.